common/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -957,6 +958,12 @@ private boolean filterDocument(
         return path;
     }
 
+    // CookieStore Provider:
+    @NonNull
+    private CBLCookieStore getCookieStore() {
+        return new CookieStore(getDatabase());
```

Comment:
Yeah... again... is there no chance of caching this thing, or something?  Do we really need to create a new one, each time?

Reply:
It's per database so we need to create a new cookie store.